content/learn/07.built-in-libraries/02.i2s/i2s.md

```diff
@@ -150,26 +165,67 @@ I2S.availableForWrite()
 none
 
 #### Returns
-the number of bytes available to write.
+size_t
+- the number of bytes available to write
 
 ### `onTransmit(handler)`
 
 #### Description
-Registers a function to be called when a block of data has been transmitted.
+Registers a function to be called when a block of data has been transmitted. This is used in conjunction with the two-argument function write(buf, len), and the callback function is called when the DMA write operation that was triggered by write(buf, len) finishes. The callback takes no arguments, which mean it cannot determine by itself if the write was completely successful or wrote only some part of the data. Partial writes need to be handled separately by the application.
+
+The callback is not triggered by the single argument version `write(val)`, only by the two argument version `write(buf, len)`. The callback can be registered once at setup of the I2S interface, and will be called for every subsequent write() -- you do not need to register it anew for every write operation.
```

Comment:
Could you please send the source of this?

Reply:
Verified by my own experiments. I had to use a lot of trial and error to get this library to work for me. I didn't find any documentation on this anywhere else, but with the current library code, the callback was most definitely not executed after a single value write when I checked just now.